standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/events/AlterTableEvent.java

```diff
@@ -40,6 +42,11 @@ public AlterTableEvent (Table oldTable, Table newTable, boolean isTruncateOp, bo
     this.newTable = newTable;
     this.isTruncateOp = isTruncateOp;
     this.writeId = writeId;
+    if (newTable.getSd() == null) {
```

Comment:
In what case newTable.getSd() can be null?

Reply:
not sure ..if the location is not changed ..then it make sense to keep it null ..to avoid extra data to be written to meta store db